libs/hwes/eventstore.go

```diff
@@ -0,0 +1,92 @@
+package hwes
+
+import (
+	"context"
+	"github.com/EventStore/EventStore-Client-Go/esdb"
+	"github.com/rs/zerolog/log"
+	"hwutil"
+)
+
+var (
+	es *esdb.Client = nil
+)
+
+func SetupEventStoreByEnv() {
+	connectionString := hwutil.MustGetEnv("EVENTSTORE_CS")
+	SetupEventStore(connectionString)
+}
+
+func SetupEventStore(connectionString string) {
+	config, err := esdb.ParseConnectionString(connectionString)
+	if err != nil {
+		log.Fatal().Err(err).Msg("invalid eventstore connection string")
+	}
+
+	client, err := esdb.NewClient(config)
+	_, err = client.ReadAll(context.Background(), esdb.ReadAllOptions{}, 1) // Test connection
+	if err != nil {
+		log.Fatal().Err(err).Msg("could not connect to eventstore")
+	}
+	log.Info().Msg("connected to eventstore")
+	es = client
+}
+
+func AppendEvents(ctx context.Context, id string, events ...Event) error {
+	es := GetES()
+
+	eventsData := make([]esdb.EventData, 0, len(events))
+	for _, event := range events {
+		eventsData = append(eventsData, event.ToEventData())
+	}
+
+	// readOps := esdb.ReadStreamOptions{Direction: esdb.Backwards, From: esdb.End{}}
+	readOps := esdb.ReadStreamOptions{}
+	stream, err := es.ReadStream(ctx, id, readOps, 1)
+	if err != nil && err != esdb.ErrStreamNotFound {
+		return err
+	}
+
+	if err == nil {
+		defer stream.Close()
+	}
+
+	// Stream not exists yet
+	if err == esdb.ErrStreamNotFound {
+		_, err = es.AppendToStream(
+			ctx,
+			id,
+			esdb.AppendToStreamOptions{ExpectedRevision: esdb.NoStream{}},
+			eventsData...,
+		)
+
+		return err
+	}
+
+	lastEvent, err := stream.Recv()
```

Comment:
I discovered that `lastEvent` is the first, not the last event of the stream when I want to dispatch the `BED_DELETEDv1` event. (after dispatching `BED_UPDATEDv1`)

This is probably because `readOps` points into the wrong direction? With line 42 as `readOps` instead, it works like a charm.

Reply:
Oh yes! The line under the comment in #42 was for testing.